src/linpred.jl

```diff
@@ -182,5 +141,24 @@ model_response(obj::LinPredModel) = obj.rr.y
 fitted(m::LinPredModel) = m.rr.mu
 predict(mm::LinPredModel) = fitted(mm)
 formula(obj::LinPredModel) = ModelFrame(obj).formula
-nobs(obj::LinPredModel) = length(model_response(obj))
 residuals(obj::LinPredModel) = residuals(obj.rr)
+
+"""
+    nobs(obj::LinearModel)
+    nobs(obj::GLM)
+
+For linear and generalized linear models, returns the number of rows, or,
+when prior weights are specified, the sum of weights.
+"""
+function nobs(obj::LinPredModel)
+    if isempty(obj.rr.wts)
+        oftype(sum(one(eltype(obj.rr.wts))), length(obj.rr.y))
```

Comment:
I think I'd go for `convert(eltype(obj.rr.wts), length(obj.rr.y))`. It's not true if the weights are Booleans but that is probably not a case we need to consider. It is also a similar issue for your version because scalar `sum` is matching the type of vector `sum`
```jl
julia> sum(true)
true

julia> sum([true])
1
```